### PR TITLE
Upgrade to 3.8.0 (actually to 3.8.1)

### PR DESCRIPTION
**Pull request by nataliecarey**

### CODE REVIEW

File: src/main/play-26/twirl/uk/gov/hmrc/govukfrontend/views/components/govukInput.scala.html [hunk 1]
```diff
@@ -65,7 +65,7 @@
 
 @inputSnippet(describedBy: String) = {
   @defining {
-    id="@id" name="@name" type="@inputType"@value.mapNonEmpty { value => value="@value" }@if(describedBy.nonEmpty) { aria-describedby="@describedBy"}
+    id="@id" name="@name" type="@inputType"@if(spellcheck.nonEmpty){ spellcheck="@if(spellcheck.getOrElse(true)){true}else{false}" }@value.mapNonEmpty { value => value="@value" }@if(describedBy.nonEmpty) { aria-describedby="@describedBy"}
```
Comment (JoPintoPaul) [inline src/main/play-26/twirl/uk/gov/hmrc/govukfrontend/views/components/govukInput.scala.html]: This can be simplified to:
@if(spellcheck.nonEmpty) { spellcheck="@spellcheck"}

Twirl templates will automatically unwrap the value inside an Option:
https://www.playframework.com/documentation/2.7.x/ScalaTemplates#Displaying-Scala-types

File: src/main/play-26/twirl/uk/gov/hmrc/govukfrontend/views/components/govukTextarea.scala.html [hunk 2]
```diff
@@ -57,7 +57,7 @@
     @autocomplete.mapNonEmpty { autocomplete =>autocomplete="@autocomplete"}
   } { autocompleteAttr =>
   @defining {
-    id="@id" name="@name" rows="@if(rows != 0) {@rows} else {5}"@if(describedBy.nonEmpty) { aria-describedby="@describedBy"}@autocompleteAttr @toAttributes(attributes)
+    id="@id" name="@name" rows="@if(rows != 0) {@rows} else {5}"@if(describedBy.nonEmpty) { aria-describedby="@describedBy"}@autocompleteAttr @toAttributes(attributes) @if(spellcheck.nonEmpty){spellcheck="@if(spellcheck.getOrElse(true)){true}else{false}"}
```
Comment (JoPintoPaul) [inline src/main/play-26/twirl/uk/gov/hmrc/govukfrontend/views/components/govukTextarea.scala.html:60]: As above re: simplification

File: src/test/resources/fixtures/test-fixtures/input-with-spellcheck-enabled/output.txt [hunk 3]
```diff
@@ -0,0 +1,6 @@
+<div class="govuk-form-group">
+  <label class="govuk-label" for="input-with-spellcheck-enabled">
+    Spellcheck is enabled
+  </label>
+  <input class="govuk-input" id="input-with-spellcheck-enabled" name="spellcheck" type="text" spellcheck="true">
+</div>
```
Comment (JoPintoPaul) [inline src/test/resources/fixtures/test-fixtures/input-with-spellcheck-enabled/output.txt:6]: Is there also a test for input with no spellcheck defined? Is it worth adding one?

File: src/test/resources/fixtures/test-fixtures/textarea-with-spellcheck-enabled/output.txt [hunk 4]
```diff
@@ -0,0 +1,6 @@
+<div class="govuk-form-group">
+  <label class="govuk-label" for="textarea-with-spellcheck-enabled">
+    Spellcheck is enabled
+  </label>
+  <textarea class="govuk-textarea" id="textarea-with-spellcheck-enabled" name="spellcheck" rows="5" spellcheck="true"></textarea>
+</div>
```
Comment (JoPintoPaul) [inline src/test/resources/fixtures/test-fixtures/textarea-with-spellcheck-enabled/output.txt:6]: As above for the use case where spellcheck is undefined.

File: src/test/scala/uk/gov/hmrc/govukfrontend/views/viewmodels/input/Generators.scala [hunk 5]
```diff
@@ -55,7 +57,8 @@ object Generators {
         classes          = classes,
         autocomplete     = autocomplete,
         pattern          = pattern,
-        attributes       = attributes
+        attributes       = attributes,
+        spellcheck       = arbSpellcheck
```
Comment (JoPintoPaul) [inline src/test/scala/uk/gov/hmrc/govukfrontend/views/viewmodels/input/Generators.scala]: You can correct arbSpellcheck back to spellcheck now, I should have done that before checking in, apologies.

File: src/main/play-26/twirl/uk/gov/hmrc/govukfrontend/views/components/govukInput.scala.html (hunk 1 above)
Comment (nataliecarey) [reply src/main/play-26/twirl/uk/gov/hmrc/govukfrontend/views/components/govukInput.scala.html]: Woohoo, I'd hoped that was possible.

File: src/test/resources/fixtures/test-fixtures/input-with-spellcheck-enabled/output.txt (hunk 3 above)
Comment (nataliecarey) [reply src/test/resources/fixtures/test-fixtures/input-with-spellcheck-enabled/output.txt:6]: Yeah, they won't be particularly clear in the diff.  One example is https://github.com/hmrc/play-frontend-govuk/blob/upgrade-to-3.8.0/src/test/resources/fixtures/test-fixtures/input-with-hint-text/input.json and there are other examples which start with `input-` in this directory https://github.com/hmrc/play-frontend-govuk/tree/upgrade-to-3.8.0/src/test/resources/fixtures/test-fixtures

File: src/test/resources/fixtures/test-fixtures/textarea-with-spellcheck-enabled/output.txt (hunk 4 above)
Comment (nataliecarey) [reply src/test/resources/fixtures/test-fixtures/textarea-with-spellcheck-enabled/output.txt:6]: As above - there are other examples which won't be clear in the diff, for example https://github.com/hmrc/play-frontend-govuk/blob/upgrade-to-3.8.0/src/test/resources/fixtures/test-fixtures/textarea-with-default-value/input.json

File: src/main/play-26/twirl/uk/gov/hmrc/govukfrontend/views/components/govukInput.scala.html [hunk 6]
```diff
@@ -65,7 +65,13 @@
...
 @inputSnippet(describedBy: String) = {
   @defining {
-    id="@id" name="@name" type="@inputType"@value.mapNonEmpty { value => value="@value" }@if(describedBy.nonEmpty) { aria-describedby="@describedBy"}
+    id="@id"
+    name="@name"
+    type="@inputType"
+    @if(spellcheck.nonEmpty) {
+      spellcheck="@spellcheck"
+    }
+    @value.mapNonEmpty {value => value="@value" }@if(describedBy.nonEmpty) { aria-describedby="@describedBy"}
```
Comment (JoPintoPaul) [inline src/main/play-26/twirl/uk/gov/hmrc/govukfrontend/views/components/govukInput.scala.html]: I notice the formatting here has changed, although the code itself looks good - is this change deliberate?

**Review by matthewmascord — COMMENTED** [review]
I think we're just missing an update to the CHANGELOG otherwise looks good to me.

**Review by matthewmascord — DISMISSED** [review]

**Review by matthewmascord — APPROVED** [review]
